frontend/Container.js

```diff
@@ -37,14 +46,50 @@ class Container extends React.Component {
     },
     extraTabs: {[key: string]: () => React$Element},
   };
+  state: State;
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      isVertical: (window.innerWidth < IS_VERTICAL_BREAKPOINT),
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener('resize', this.handleResize, false);
+    this.setState({
+      isVertical: (window.innerWidth < IS_VERTICAL_BREAKPOINT),
+    });
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.handleResize);
+  }
+
+  handleResize = (e: Event) => {
+    if (!resizeTimeout) {
```

Comment:
React synthetic events get reused. I don't think it's safe to pass them around. Could you read target here and then pass just the target?